### ISSUE TITLE
Terms are not correctly collected when `disablePathToLower` set to true with custom `_index.md`

### ISSUE THREAD

**Issue by cmpute** [gohugoio/hugo#10630]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.109.0-47b12b83e636224e5e601813ff3e6790c191e371+extended linux/amd64 BuildDate=2022-12-23T10:38:11Z VendorInfo=gohugoio
</pre>

### Does this issue reproduce with the latest release?

Haven't tried but I believe so.

### Description

The taxonomy terms are not correctly associated with the taxonomy page if `disablePathToLower` is true. This problem can be reproduced using [the `exampleSite` of the theme stack](https://github.com/CaiJimmy/hugo-theme-stack/tree/master/exampleSite).

Under the default settings, the posts of category `Test` are correctly attached to the test page (`/categories/test`).
![image](https://user-images.githubusercontent.com/9154441/213506849-17a22d31-54e4-4b69-b144-efd0963ca9d6.png)

However, if `disablePathToLower` is set to True, then the path of the category page becomes `/categories/Test` and this page contains no actual posts.
![image](https://user-images.githubusercontent.com/9154441/213506508-8ed4761a-71a7-4fe7-95ce-a6998a761921.png)

So in summary, I guess the `disablePathToLower` option messed up the collection process of taxonomies.

#4183 might be related. And I'm aware of #9171.

I use this option because of historical considerations, I don't want to mess up the SEO of my website. I believe I have to migrate to all-lower-case URL path, because this issue is critical, and it might not be fixed soon due to #9171.

**Comment by cmpute** [comment]
@jmooring You can produce the problem by adding a custom title to one of the tags in your repo. For example, create a `content/tags/Tag-A/_index.md` as follows:

```markdown
---
title: Tag A
---
```

As long as you have a separate `_index.md` for a tag / category, there will be no posts under that tag / category.

**Comment by jmooring** [comment]
@cmpute Confirmed. Please update the issue title.
